### ISSUE TITLE
Chicago Style Troubles: double place, trans. before.

### ISSUE THREAD

**Issue by urshofer** [seboettg/citeproc-php#113]
# Please follow the general troubleshooting steps first:

- [x] I read the README and followed the instructions.
- [x] I am sure that the used CSL metadata follows the [CSL schema](https://github.com/citation-style-language/schema/blob/master/csl-data.json).
- [x] I use a valid CSL stylesheet

Thanks for your great Work!

I have the following Issue, using chicago-fullnote-bibliography-16th-edition.

This is the metadata: 

```
{
        "type": "book",
        "id": "williams_daniele_2019",
        "citation-label": "williams_daniele_2019",
        "issued": {
            "date-parts": [
                [
                    2019
                ]
            ]
        },
        "publisher-place": "[Basel]",
        "publisher": "Birkhäuser",
        "title": "Daniele Barbaro’s Vitruvius of 1567",
        "translator": [
            {
                "given": "Kim",
                "family": "Williams"
            }
        ]
    }
```

Is rendered like this:

```, trans.Williams, Kim. Daniele Barbaro’s Vitruvius of 1567. [Basel]: Birkhäuser, 2019, [Basel]: Birkhäuser, 2019.```

Should be rendered:

```Williams, Kim, trans. Daniele Barbaro’s Vitruvius of 1567. [Basel]: Birkhäuser, 2019.```

(Test made with https://editor.citationstyles.org/visualEditor/)

Any hints - I'm quite new to csl, worked with BibLaTex before.

Thanks!

**Comment by urshofer** [comment]
So, it's in this macro:

```
  <macro name="translator">
    <names variable="translator">
      <name and="text" delimiter-precedes-et-al="never" delimiter-precedes-last="always" name-as-sort-order="first"/>
      <label form="verb-short" prefix=", "/>
    </names>
  </macro>
```

Where `, trans.` is always preceeding, which is only correct if the author/editor is existing.

**Comment by urshofer** [comment]
I further pinned down the issue.

In the csl file there are several constructions like this:

```
<choose>
    <if type="speech" variable="publisher publisher-place" match="any">
        <text macro="issue"/>
    </if>
</choose>
```

If I rewrite the construction like this

```
<choose>
    <if type="speech">
        <choose>
            <if variable="publisher publisher-place" match="any">
                <text macro="issue"/>
            </if>
        </choose>
    </if>
</choose>
```

It works as expected.

It looks like the scope of `match="any"` is different between the standard implementation as seen on https://editor.citationstyles.org/visualEditor and citeproc-php. First is testing both (type && variable||variable||variable), latter probably (type || variable || variable …) or ignoring type at all.

**Comment by jmvezic** [comment]
@urshofer I'm having the same issue. Is this a problem with the CSL file or with citeproc-php's handling of the CSL file? How come citeproc-js doesn't have this same issue?

**Comment by urshofer** [comment]
I guess the bug is on citeproc-php's side. I did not have time to investigate further, so
i changed the csl file a little bit to make it run with citeproc-php.

**Comment by seboettg** [comment]
Hi @jmvezic!
Please check if the issue is solved with version 2.4.2.

**Comment by jmvezic** [comment]
Yeap, solved with 2.4.2!

As an aside, Composer wouldn't pick up the 2.4.2 version for some reason, even with `^2`. I had to specifically say `2.4.2`.
